build.sbt

```diff
@@ -65,6 +67,15 @@ def crossSetting[A](
     case _ => Nil
   }
 
+def publishCrossVersion(scalaVersion: String): CrossVersion = {
+  // similar to `CrossVersion.binary`
+  // but for scala3 write both major and minor instead of just "3"
```

Comment:
Hm, can you elaborate on why we need this? I'm getting lost in the whole 3.1/3.0 situation so would appreciate some extra information.

I thought that we can publish mdoc with whatever Scala 3 version we want, and if it's a `minor` below the end user's version, they won't have to upgrade?

Reply:
To make sure I'm following, this will publish `_3.1` and `_3.0` artifacts? If so, and maybe a silly question, but then how would a user of mdoc include this in their build? Since it's not the default `_3` that the build tool would look for? Or am I totally missing something.

Reply:
@ckipp01 it's handled by [`MdocPlugin`](https://github.com/scalameta/mdoc/blob/d4e4b8c6303793b7a5c0040b4e35ae64e0140c4a/mdoc-sbt/src/main/scala/mdoc/MdocPlugin.scala#L90-L99).

Reply:
But that assumes that someone is using it through the plugin. What if I'm just using mdoc as a library?

Reply:
@keynmol  That's indeed a #525 . I faced the same problem in [metals](https://github.com/scalameta/metals/pull/3212)

I don't think there aren't a lot of direct users of mdoc dependency to worry about non-std artifact name.
I agree that's a bit inconvenient but they can declare it using full name like: `"org.scalameta" % "mdoc_3.1" % $version`

Reply:
I am indeed one of those users of mdoc without plugin, so this affects me directly in several internal projects :) 

But on another note, I still think we should investigate alternatives or indeed the root cause.

This goes against the way Scala 3 artifacts are supposed to be published, and while I understand blocking metals release is unfortunate, this will produce artifacts which the rest of the ecosystem will not know how to treat, as Scala 3's binary artifact version is supposed to be `_3` and that's it.

Reply:
👍 awesome